### ISSUE TITLE
Open an RFC to discuss decision to revert License and remove Jamie

### ISSUE THREAD

**Issue by hannahhoward** [lerna/lerna#1639]
While choosing to close https://github.com/lerna/lerna/issues/1635 @TheLarkInn writes:

```
Ben I'm going to close this issue, in favor or a more structured, RFC style of a proposal for what decisions you are questioning. Thanks! We don't have a template RFC format, or spec template, but hopefully looking to have one once we catchup and dust settles.
```

It's not clear what specifically such an RFC would look like, and what it's format would take. Since then no issue has been opened.

This issue is simply to register there is dissent among a portion of Lerna users about the decision of the core team to revert Jamie's license change and remove him from the project until an RFC can be opened. 

It is precisely the time when people are paying attention when discussion is necessary, when a wider group of perspectives can be heard.

**Comment by shushugah** [comment]
My simple question, given the lack of due discussion for the previously MIT license modification, would the maintainers be open to a new PR with legally more robust language along with input from the community?

I definitely think the previous process was rushed, but the core idea is worth exploring.

**Comment by catcher-in-the-try** [comment]
I think a good idea would be to close this issue, cool down for a week or two, and then open the RFC.

**Comment by hannahhoward** [comment]
@catcher-in-the-try I disagree. It is relevant for two reasons:

1. There is significant dissent about the decision to revert the changes in the Licenses and remove Jamie, and an RFC was suggested as a mechanism for discussion, but does not exist. This is a placeholder to simply register there is unaddressed dissent and a promise of a future conversation.

2. There seems to be a fear of having a discussion while things are "heated" -- I would argue that this is a good time to have discussion, as it's the time when there is likely to be the widest set of perspectives and the most voices are likely to be heard.

**Comment by benwiley4000** [comment]
Calls for "civility" and "cooling down" are invariably intended to cease discussion, not enhance it. Hannah's right - the time to discuss is now.

**Comment by nls0** [comment]
If you want to restrict who can use lerna, then lerna can no longer be called free; it will be proprietary. Please let everyone know with advance notice so they can make preparations to move away from it if it's going to become proprietary software.

**Comment by catcher-in-the-try** [comment]
I made you all a [fork](https://github.com/catcher-in-the-try/lerna-mit), case closed, you can change the license to non-free.

@hannahhoward  I do doubt the ability of humans to make rational decisions during heated discussions. In law, this fact is even recognized as a valid defense for making bad and illegal choices.

**Comment by benwiley4000** [comment]
I don't think anyone's particularly heated, just excited to talk about this. 🙂

**Comment by darthtrevino** [comment]
I see these as two separate issues. If the community and the maintainers want to use a modified MIT license, that is fully within their right. However, Jamie's behavior was plainly abusive and in violation of the CoC.

**Comment by catcher-in-the-try** [comment]
@benwiley4000 Interesting that you don't find this heated, I see people going into random repositories and opening issues calling everyone nazis. 

That looks to me as if someone needs to take a break from the news and politics and regain the ability for reasonable discussion.

**Comment by hallister** [comment]
Jamie's removal was a necessity and shouldn't be open for debate. Blocking people from contributing because they disagreed, which eventually lead to him blocking everyone but contributors, attacking companies/individuals, claiming community tools (like Babel) are "his" is overwhelming evidence that he violated the CoC.

@benwiley4000 Calls for civility are perfectly apt after an issue like this. Suggesting that asking people to "be nice" is an attempt at ending discussion is one of the strangest stretches I've ever heard. As someone that called maintainers cowards for reverting the changes and removing Jamie, I can get a strong sense of why you'd argue that point, however.

As for this issue, @TheLarkInn made it pretty clear that they'd like to wait for the dust to settle prior to a formal RFC. That makes sense. Right now Lerna is a target for people only interested in politics, and once the dust settles they will likely have something new to chase, and the people actually interested in the project can contribute to the suggested RFC.

**Comment by hannahhoward** [comment]
"Right now Lerna is a target for people only interested in politics" -- I am a professional coder. I am interested in code. I am also interested in politics, FWIW. Given the original issue was a political one, it seems particularly important that a wide variety of political perspectives are included. To wait until there is a safe space for those "only interested in code" excludes a variety of political views, as the idea that politics and code can and should be separate is itself a political position that many disagree with.

I am not personally prepared to say Jamie's removal is a settled issue because the announcement he is being removed does not include any accounting or transparency on what specific violations the maintainers feel occurred. But I agree it is a separate question, except in as much the maintainers announced it in the same PR where they reverted the license change, suggesting they consider the two issues intertwined.

**Comment by hannahhoward** [comment]
@catcher-in-the-try making a fork suggests you are engaging in this discussion in bad faith, and would simply like people who have a different viewpoint to go away.

**Comment by hallister** [comment]
@hannahhoward 
> Given the original issue was a political one, it seems particularly important that a wide variety of political perspectives are included. To wait until there is a safe space for those "only interested in code" excludes a variety of political views, as the idea that politics and code can and should be separate is itself a political position that many disagree with

If the broader open-source community wants to discuss the politics of this, they should do so in a forum that makes sense. The Lerna repo isn't that forum. This repo is intended for end-users and maintainers of the project. As such, decisions about the goals and motivations of the project should be made by the Lerna community.

> I am not personally prepared to say Jamie's removal is a settled issue because the announcement he is being removed does not include any accounting or transparency on what specific violations the maintainers feel occurred. 

I realize some people feel like open source maintainers need to write books for every decision they make, but I disagree. It was decided by multiple maintainers that he violated the CoC. That's subjective, absolutely, but debating the issue to death isn't suddenly going to give us objectivity.

**Comment by mAAdhaTTah** [comment]
The purpose isn't objectivity, it's transparency.

**Comment by benwiley4000** [comment]
@hallister Justin your implication that commenters have no relation to Lerna is unsubstantiated. I for one am actively involved in the ecosystem surrounding Lerna and have been preparing to add Lerna to a project I maintain, although I am now considering alternative options.

**Comment by hallister** [comment]
> The purpose isn't objectivity, it's transparency.

Let's test if that's the actual goal

> Examples of unacceptable behavior by participants include the use of sexual language or imagery, derogatory comments or *personal attacks*, *trolling*, public or private harassment, *insults*, or other unprofessional conduct.

(Emphasis added by me)

> personal attacks... insults

Per https://github.com/palantir/blueprint/issues/2877

>> Palantir employees are racist

> trolling

* Blocking people on the lerna repository that politely called him out.
* Blocking all non-contributors on the repo from posting issues/comments/PR's

Per https://github.com/lerna/lerna/issues/1628

* Suggesting he may not do a major version bump

This is in addition to his horrible conduct in every issue posted regarding adding his company restrictions to the MIT license. That's pretty transparent. That's pretty objectively trolling, personal attacks and insults. Now the question is, do you really want transparency, or are you looking for a fight?

**Comment by hallister** [comment]
@benwiley4000 
> Justin your implication that commenters have no relation to Lerna is unsubstantiated. 

I made no such implication. I said, quite explicitly, that Lerna is currently a target for those disinterested in the project and overly interested in the politics. By waiting for the dust to settle, we can ensure that any RFC is targeting the actual community. I'm not suggesting that everyone that supported Jamie is a troll, I'm saying that right now Lerna is going to attract them. Wait a few days and they'll be busy trolling someone else so we can get active interest in the discussion, including those that disagree with reverting the change.

**Comment by jwietelmann** [comment]
Re: "cooling off" and "people interested only in politics"...

I'm seeing an extremely ill-informed opinion being bandied about that implies the current state of open source is apolitical. FSF was founded on some radically collectivist ideals. OSI on more radically libertarian ones. "Open source principles" are what they are because people planted the flag and did the work to make them that way. The reason this project is MIT instead of GPL in the first place is because of decades of ideological debate and changes in community expectations.

Planting a different flag and doing a different thing is not more or less political than the status quo. To suggest otherwise is ahistorical nonsense. It's not more political; it's merely more participatory.

I'm simultaneously intrigued by the effort to change the license and skeptical of what its effects might be. Maybe reverting was a good idea! It would have for sure been _more interesting_ to see the experiment run its course, but I understand the fear to the health of the project that makes someone walk it back.

Of course this attracted wider attention. Of course people who were otherwise mostly foreign to lerna were intrigued and came running. You were doing something mad and exciting and extremely political. A big project was finally taking an ethical question and putting it front and center and saying "Yeah, it's time to talk about this."

Post-blowback, you don't wanna be political pioneers anymore. And who does? A) It's hard, and B) that's not how most of us want to spend our OSS time.

If you're overwhelmed, frazzled, can't handle being the center of this ethical discussion you started, and just want to be left alone to code now... You could just say so?

**Comment by hannahhoward** [comment]
@hallister transparency for a decision comes from the people who made the decision. Right now what we have is:

```
Despite his numerous (and appreciated) contributions in the past, it has been very clear for quite some time now that he has decided to cease making constructive contributions to the Lerna codebase as well as actively and willfully disregarding the code of conduct that he himself added to the project.
```

You and I can quibble indefinitely over whether the things you posted are actually CoC violations. Something of the things you posted don't meet my standards personally.

**Comment by evocateur** [comment]
Please have patience with me. I don't have a lot of spoons right now, between dealing with an unprecedented (for me) deluge of Twitter notifications and staving off suicidal ideation.

**Comment by shushugah** [comment]
FWIW @jwietelemann is right that Lerna is getting much attention, eg see this Vice motherboard news article. I did not know about this project myself till I saw the debate. 
https://motherboard.vice.com/en_us/article/8xbynx/major-open-source-project-revokes-access-to-companies-that-work-with-ice

**Comment by TheLarkInn** [comment]
@hannahhoward 

> This issue is simply to register there is dissent among a portion of Lerna users about the decision of the core team to revert Jamie's license change and remove him from the project until an RFC can be opened.

Thank you for speaking up and registering your dissent and representing the dissent of like-minded Lerna users. 

As an open source project we have every right to provide no reason for his removal. 

However, as stewards of transparency, trust, and care for our users and the lerna ecosystem, we will provide clarity.

According to the current Lerna Code of Conduct:

> Examples of unacceptable behavior by participants include the use of sexual language or imagery, derogatory comments or personal attacks, trolling, public or private harassment, insults, or other unprofessional conduct.

Additionally our rights as Org Members of the project for handling unacceptable behavior which do not align with the current Code of Conduct.

> Project maintainers have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct. Project maintainers who do not follow the Code of Conduct may be removed from the project team. 

These are the segments of the current lerna Code of Conduct that justify our decisions made to revoke all of James involvement and ownership privileges over this GitHub Organization. 

In multiple GitHub issues (if you need me to cite them, I will do so but in additional comments or edits from this response), on Twitter [important to note he claims ownership of lerna in this context making him profesionally represented and in the capacity of a maintainer] acts in a very unprofessional, rude, and harassing manner.

These also have been occurring since July which @evocateur official states in our license change PR #1633 that we made a mistake in this regard to not address earlier violations in a swift and timely manner. 

As a core team, going forward, we want to not only protect the interest of the project itself, but also the transparency of the project. And that means we need to adopt a much less vague, less interpretive, and more structured Code of Conduct. For that we have open up #1636 per request (I on twitter officially asked Coraline to create this issue).

On behalf of the core team who is all actively monitoring this issue, I hope it provides the clarity that you are seeking. Thank you for sticking up for a transparent and responsible process.

**Comment by benwiley4000** [comment]
@TheLarkInn I think where we need additional clarity is:
* Why did you decide to revert the license change despite maintainers having originally been involved in approving it?
* Why did you choose to link Jamie's dismissal with the license change?
* Why did you open a pull request to remove your own employer (Microsoft) from the restricted list?

**Comment by nls0** [comment]
https://github.com/palantir/tslint/issues/4132
https://github.com/palantir/tslint/issues/4141
https://github.com/palantir/blueprint/issues/2870
https://github.com/palantir/blueprint/issues/2876
https://github.com/palantir/blueprint/issues/2877

Why did he keep opening so many issues, harassing other people and why did he repeatedly refer to these free software projects that many people have contributed to as "my tools"? Is all the code his property? Are all the contributions by everyone else worthless?

"I kinda hope they do try to keep using **my** tools though" - From the motherboard article, emphasis added
"Also, stop using my tools (such as Babel)" - from /palantir/tslint/issues/4132

Babel is his property as well? Can someone help my understand why these are his property? Thanks.

**Comment by mAAdhaTTah** [comment]
@nls0 Those questions aren't relevant to the thread. You'd have to ask Jamie himself.

**Comment by nls0** [comment]
@mAAdhaTTah Those questions are relevant to the thread. Read the title of the thread.

**Comment by benwiley4000** [comment]
@TheLarkInn:

> This decision was made by @evocateur and his resoning is very clear in #1633.

I disagree that this is very clear:

"the impact of this change was almost 100% negative, with no appreciable progress toward the ostensible goal aside from rancorous sniping and harmful drama."

The claims made here are not supported by evidence or arguments.

> I did not link Jamie's dismissal. @evocateur explained his reasoning in #1633

This was the plural "you" here, as in "you, the maintainers." In any case, I still don't understand why Jamie's dismissal was linked as part of the same announcement, clearly implying that his dismissal had to do with his instruction of the modified MIT license.

> I thought Microsoft was being treated unfairly and wanted to set the story straight.

Respectfully, given the conflict of interest between your role as a maintainer and your role as a Microsoft employee, I think you should have sat out of that decision. It certainly appears to many as though your loyalty to your employer influenced the revert back to the old license.

**Comment by evocateur** [comment]
> It certainly appears to many as though your loyalty to your employer influenced the revert back to the old license.

It did not. Please stop repeating this baseless canard.

**Comment by benwiley4000** [comment]
@evocateur this is transparently absurd. You can't just say "I'm not influenced by the company who pays me" and thus make it so. I sometimes laugh at the Twitter refrain of "tech workers need to be trained in ethics," but perhaps it's true.

**Comment by evocateur** [comment]
@benwiley4000 If you're looking for "rancorous sniping and harmful drama" you might try using a mirror.

**Comment by evocateur** [comment]
Explain how me, the person who made the decision, not employed by Microsoft, was secretly paid by Microsoft to ruin open source. Should be an amusing story.

**Comment by benwiley4000** [comment]
You explicitly stated on Twitter that changes were being made after speaking with Sean, a Microsoft employee. If this isn't what happened, then what did happen is certainly hard to follow.

**Comment by evocateur** [comment]
I made the decision after speaking with Sean. It is possible to talk to somebody and then make an independent decision. If that's hard to believe, it's not my problem.

**Comment by hallister** [comment]
@benwiley4000 
> The claims made here are not supported by evidence or arguments.

Feel free to show appreciable change as a result of this. At Amazon someone sent an email that said "Don't use lerna above version v3.2.1". I could just feel ICE crumbling under the pressure.

> Respectfully, given the conflict of interest between your role as a maintainer and your role as a Microsoft employee, I think you should have sat out of that decision. 

Yes, only people with no vested interest in this can be involved in the decision making.  That makes a ton of sense.

**Comment by hallister** [comment]
> I personally think this action has definitely moved the needle on this issue.

Because a guy among 10s of thousands was going to put up the big fight to stay off the list, and now can't because it doesn't exist anymore. I suppose, instead, he can now argue to not have contracts with ICE, but then this silly move wouldn't get any credit.

> Court judges recuse themselves from decisions involving conflicts of interest all the time. The idea that one might have such a conflict and thus should not be involved in the decision is not particularly far-fetched.

Jamie already had a conflict of interest in targeting specific companies he didn't like. Your argument here would imply only people that are independent should play a role in the decision-making. Except nobody is independent.

**Comment by mAAdhaTTah** [comment]
@hallister I'm inclined to view this like a roach: one person visible is indicative of a lot more you don't see. The point is that this caused at least one person to publicly say "this resulted in me fighting to oppose ICE contracts." I believe that's resulted in a lot more people than just him doing similar things. I know I've had a few conversations with people who hadn't previously considered the issue before this.

> Jamie already had a conflict of interest in targeting specific companies he didn't like.

That's not a conflict of interest. Judges don't recuse because of what they believe; they recuse themselves because they have a material interest in the result of the ruling, e.g. they've invested stocks in the companies involved. You're correct that nobody is independent, because no one is objective, but conflicts of interest and lack of independence are not synonymous.

---

All of that said, we're getting off topic from the purpose of the thread. I'm happy to continue this offline if you'd like.

**Comment by dmfay** [comment]
@evocateur I know you're dealing with a lot more than you expected right now but I'm afraid that as the maintainer of a project with a decently-sized community it really is your problem. I don't envy you it. I frankly don't know if it's even possible to clear it up completely at this point.

At a charitable reading, Sean seems to have jumped in feet-first without thinking about what it looks like for a Microsoft employee to get involved in a situation which affects Microsoft's access (particularly given that Microsoft is in fact well known to have an active engagement with ICE, & that the original license change didn't allow for equivocation about legacy systems). To be clear, this is how _I_ think it went down too. I've met Sean and he cares a lot about open-source communities and open-source projects. But there are less charitable readings which are equally plausible from what's publicly known, or even just public on GitHub vs public on Twitter. You know as well as I do that "big MS puts the screws to open-source project" is a headline going back years and years, and Sean's involvement in _any_ capacity with this chain of events means you've inadvertently leaned into that old stereotype.

**Comment by hannahhoward** [comment]
@TheLarkInn

I wish I could say your response made me feel discussion should be locked on this issue and consider it resolved. Unfortunately, it has the opposite effect for me.

What I can see is:
1. A change was made to the license that spoke ill of Microsoft was made that @evocateur approved of and merged 
2. You, a Microsoft employee, attempted to remove Microsoft, and only Microsoft, from the list of companies in said change (I didn't know that part)
3. @evocateur then spoke to you, a Microsoft employee, at length, after which the change was reverted. Speaking to you at length was specifically called in the decision to revert.

I am going to assume best intention that you genuinely feel calling out Microsoft was unfair, and you genuinely sought changes because of that, and not in an effort to protect your employer. However, as an employee of one of the companies that were called out in the change, clearly the ethical decision would be for you to recuse yourself from discussions of this issue completely. There's really no way in which you're a vaguely objective party -- or even if you are it's pretty reasonable for other people not to perceive you that way.

My original issue (an RFC to reconsider the License) is not addressed, and moreover, I honestly suggest the right course for you all is for @TheLarkInn to recuse himself from discussion going forward.

* When I say "ethically" I purely speaking from what I think is right and in terms of staying impartial whenever one is given a governance role. My previous experience to being in tech is on non-profit boards, where for example, you would recuse yourself from any decision that might affect you personally as well as the organization. Obviously there is no legal duty in this case for an informal organization like Lerna *

**Comment by benwiley4000** [comment]
I am in agreement that Sean does great work for the open source community and I'm really happy about what he's done with webpack. I know he cares a lot and means the best. In fact I'm confident every maintainer involved here cares a lot and had the best intentions.

That said, I stand by Hannah's comment above.

**Comment by indrora** [comment]
As an outsider (I've only ever just used Lerna once, ended up coming by because I smelled fire yesterday)

This thread, from the outside, is a train wreck. It was opened -- with good intentions -- by someone external, when it should have been opened by someone in the team. It has turned into a means for @TheLarkInn to be beaten with; at best, this is poor form, at worst, it's harassing the man. 

I agree with @evocateur w/r/t baseless canard. I'd honestly hope for better out of the free software community, but I guess the 90s never ended, and it is easier to accuse someone of horrible things than to give them benefit of the doubt. If the Lerna team members think he should recuse himself *then that is their choice*, but a lynch mob here serves no purpose other than to harass him. 

I'm going to suggest that this thread be restricted to members of the Lerna team; shy of that, closing this thread and opening one which is restricted to Learna team members. I agree transparency is important -- *especially* in @TheLarkInn 's case. Right now, however, this is an issue for the lerna team to hash out and find a point where they feel it's settled to where the general internet can come and see what has happened, voice their thoughts, and a final choice can be made.

**Comment by csidell-playq** [comment]
It's funny how much of the argument is going on trying to express that you all have the same political views but you're too ignorant to really listen to each other. It's always jumping to conclusions that there must be conflicts of interest and conspiracy theories.

At what point are people going to realize the arguing you're doing is for the most part wasting time trying to explain to someone, *who agrees with you*, why they should agree with you. Law is not based on feelings, licenses are law, licenses are not meant to be interpreted by feelings. Whatever they may be the license has to be rigid and precise, any lack thereof leaves the license weak and unenforceable.

This isn't so much directed to the Lerna team as much as it is to every single person who is trying to ensure they're *on the right side of history* and bring the crusade Jamie was trying to create.

**Comment by Sieabah** [comment]
It's funny how much of the argument is going on trying to express that you all have the same political views but you're too ignorant to really listen to each other. It's always jumping to conclusions that there must be conflicts of interest and conspiracy theories.

At what point are people going to realize the arguing you're doing is for the most part wasting time trying to explain to someone, *who agrees with you*, why they should agree with you. Law is not based on feelings, licenses are law, licenses are not meant to be interpreted by feelings. Whatever they may be the license has to be rigid and precise, any lack thereof leaves the license weak and unenforceable.

This isn't so much directed to the Lerna team as much as it is to every single person who is trying to ensure they're *on the right side of history* and bring the crusade Jamie was trying to create.

**Comment by hallister** [comment]
I work at Amazon, should I also recuse myself from the discussion? This idea that employees of the company are too tainted to *discuss* the future of a project is absurd. If we want to debate the merits of this decision, which is how it should have gone in the first place, let's do so. But asking anyone to recuse themselves from *discussion* is the literal opposite of everything opensource is supposed to mean.

This entire situation wouldn't exist if it weren't for some poor choices, and at least one of the maintainers realized that. Had this rash choice not been made in the first place, nobody would be here demanding that discussion take place and an RFC on taking Lerna to a non-opensource license would exist. I'm fine if the maintainers want to go that direction, but if I were in their position, my vote would be to close this issue and everybody go back to building software which is *supposed to be the point* of projects like this.

But now that pandoras box has been opened, everyone with an agenda is here to fight for it, so much so, they want to silence anyone with an opinion that differs because "ethics" or whatever silly nonsense they can use to make this issue an echo chamber

**Comment by Sieabah** [comment]
@darthtrevino If you can't see the difference between that and this situation then your opinion isn't that valid.

**Comment by darthtrevino** [comment]
Of course they're different - I'm pointing out that pooh-poohing "ethical concerns" as silly nonsense is is trollish.

**Comment by hallister** [comment]
@darthtrevino You completely missed the point here. Silencing someone in the name of ethics isn't ethics. @hannahhoward explicitly stated that @TheLarkInn shouldn't even be a part of the *discussion* for ethical reasons. That's abusing something good (ethics) for a clear agenda. That's after arguing that this discussion has some degree of immediacy so that a broad variety of opinions can be heard. Except, obviously, the opinions of the people that they are well aware disagree with them.  

It's either dishonest or naive. I choose to assume the latter as that requires the least assumption. But you're correct I'm going to call out anyone that tries to abuse something as meaningful as ethics, whether or not it was out of malice or poor judgement.

**Comment by richseviora** [comment]
Hey so I don't work at any of the companies involved or know any of the maintainers personally so I don't have any potential ethical dilemmas. Here's my thoughts:

The key part of the whole conflict of interest idea is that you have to be making a decision on behalf of a 3rd party, and the conflict would give rise to the appearance that you're being unduly influenced by this secondary consideration. I believe @evocateur made the decision. @TheLarkInn had no conflict of interest, he's providing his input as a major stakeholder. As long as we all understand his role (and I think we do) then I think we'd be far richer for his input than not.

This also makes the assumption that the rash & unannounced decision to change the license was in the project's best interest, where one could easily argue that it was not for all the obvious reasons (community split, legal viability, no community feedback solicited, etc).

Generally the notion that someone's involvement with an affected organization means that they can't speak from their expertise would cause all sorts of problems if it were applied to government. As much as lobbyists have (deservedly) earned a bad reputation, they also provide vital guidance to governments so they can make decisions that aren't destructive. 

So let's get everyone's best opinions in here, and then let the maintainers make their opinion.

**Comment by richseviora** [comment]
> That's not a conflict of interest. Judges don't recuse because of what they believe; they recuse themselves because they have a material interest in the result of the ruling, e.g. they've invested stocks in the companies involved. You're correct that nobody is independent, because no one is objective, but conflicts of interest and lack of independence are not synonymous.

I think this is mostly not accurate. As a maintainer Jamie had the ability to make changes and to persuade the other maintainers. When he made this decision to pursue goals that are unrelated to the project's unstated goal (make multi-package repos easy) he was arguably engaged in his own conflict of interest. 

That said if he was the only maintainer and decided that he was changing the project's goals, then he could not be in a conflict of interest because his goals and the project's were always aligned.

**Comment by krainboltgreene** [comment]
Just for clarification, @TheLarkInn is not the only Microsoft Employee who has been a part of this ongoing discussion which makes this even more distressing.

**Comment by hallister** [comment]
@krainboltgreene The only thing distressing is that this entire event took place. You do realize that the big tech companies even let us think for ourselves?

I work for Amazon and not a single project in our org uses lerna. I do, personally. This wasn't about bias, this was about virtue signaling, particularly on such a terrible platform for it. Frankly I'm not sure where else this discussion has to go, since it's just boiling down to "if you work for any of the companies listed you shouldn't have input".

**Comment by lhorie** [comment]
Apologies if this is not an appropriate channel to voice opinions about the contents of the hypothetical RFC mentioned in the title of this issue, but assuming it is, I want to voice that I strongly disagree with the idea of applying Jamie's modified MIT license to Lerna. Not because it has anything to do with ICE, but due to the reasons cited [here](http://esr.ibiblio.org/?p=8106) - TL;DR: it would make Lerna fundamentally non-open-source.

I think that people that want a fundamentally non-open-source licensed monorepo management project should be prepared to fork Lerna, find maintainers who agree with said non-open-source license, and potentially pay for its ongoing maintenance, rather than be questioning the decision of the mainline Lerna maintainers (namely, @evocateur who created the PR, as well as @TheLarkInn and @hzoo who approved it). They are entirely within their rights to do as they please with Lerna. 

I'd also like to remind everyone involved that an RFC doesn't necessarily mean that dissenters will somehow be able to articulate an overwhelmingly convincing argument and magically overturn the maintainers' decision to retain the unmodified MIT license (or the decision to remove Jamie as a maintainer). For better or worse, the maintainers are well aware now of the implications of what has transpired and trying to nitpick perceived flaws of logic is, at best, splitting hairs at this point.

Lastly, @evocateur even if there a deluge of people accusing of you supporting political positions that you do not subscribe to, and that is taking an emotional toll on you, please remember that there are also those of us that understand your position on ICE, on OSS and on being on a publicly vulnerable. I personally believe your standing by the MIT license was the right move, and I'm not alone in thinking this. Stay strong.

**Comment by ajhool** [comment]
For left-leaning software devs who are interested in constructively affecting political change, please check out Tech For Campaigns at https://www.techforcampaigns.org/ , which pairs volunteer expert technologists with Democratic campaigns (even at the state and local level). I'm not affiliated.

Using Palantir's open-source tools does serve as promotion / brand-awareness for a company that you might think has abhorrent practices and I do think that there should be room for opening tickets to request a different vendor on that basis alone. Or even a ticket on the open-source provider's repo to request transparency. But there is obviously a reasonable and an unreasonable way to go about that -- opening those issues was a clear waste of good reputation by somebody who could have made real waves with a more level-headed approach. Perhaps community input before big decisions is not a burden but actually serves to prevent bad decisions

**Comment by robbyoconnor** [comment]
This whole thing has turned into a tire fire. "If you don't support us entirely, you're against our cause" is the most toxic thing ever. In my view, this hurts lerna's reputation. James Kyle's behavior is absolutely disgusting and I support his removal from this project. I'd also support levying a complaint with GitHub since he definitely violating their TOS, but I am not going to do that myself, but I definitely encourage those who were targeted to do so. 

I do support what was initially done, what needs to happen is that the license is crafted in a way that spells out everything. Make sure that it can't be challenged as being ambiguous. Ambiguity is your enemy here. I am also rather distressed that I was personally emailed and harassed and accused of supporting the disgusting policies of ICE. ICE needs to go and it can't be gone soon enough. 

Rather than being toxic -- which doesn't advance your cause -- it hurts your credibility because nobody wants to listen to you, consider trying to lobby the companies. I wish people would learn that the kind of stuff James Kyle pulled is absolutely disgusting and hurts the cause, which I support...it was just botched epically.

**Comment by robbyoconnor** [comment]
Also, one other thing: Microsoft owns GitHub -- maybe make the change after moving to something like GitLab :thinking: which isn't owned by a company you explicitly excluded in your license...

**Comment by hannahhoward** [comment]
It's frustrating to engage in good faith and assume best intent of others, and get accused of, in the last several comments: silencing, virtue signaling, abusing ethics to support an agenda, and engaging only to insure I'm on the right side of history (whatever that means), attempting to make an echo chamber, etc.

I stated my own personal view of the ethical way to proceed most likely to maintain the trust of the community. I specifically posited it as a suggestion. Perhaps ethics is the wrong word -- maybe I mean "best practice". If it sounded judgemental I'm sorry.

The echo chamber accusation sounds seems uniquely weird given this thread is 55 comments worth of multiple perspectives and disagreement.  I personally don't think it's a trash fire, but YMMV. It's specifically the kind of discussion I feel is needed, with people having to think about these questions discuss and disagree.

**Comment by robbyoconnor** [comment]
@hannahhoward -- I'm regretting ever even entering this discussion as it's just caused me grief and emotional distress. I can't even imagine how @evocateur is feeling right now -- he has my support 100% -- some of us are rational. Most aren't necessarily against what happened, but _HOW_ it happened. Even still, GitHub would be in their right to terminate James' account since he definitely violated the GitHub TOS. Bullying and harassment is not how you accomplish things. This is wrong and just horrible. I know there are others who got a similar email to the one I received and if you are feeling any distress, you have  my sympathies and such. This is a dumpster fire right now.

**Comment by robbyoconnor** [comment]
With the exception of a few trolls -- we all agree mostly...with a few people that have some conflicts of interest but I don't feel that they should be excluded from the discussion because of those.

**Comment by richseviora** [comment]
@hannahhoward in case it wasn't clear, while I disagree with your recommendation I'm confident it was made with the best of intentions and the goal of facilitating the best solution for all. :)

**Comment by SRGOM** [comment]
Can you social justice warriors stop using (and hence promoting) this platform owned by a major ICE collaborator? Remember just like bad news is good news in show biz, any users and discussions on githib are (or will be in 4 momths) good news for MS. Only you enlightened people can kill Microsoft and consequently ICE in one fell swoop by altogether stopping using github. also somebody pointes out that except launch pad all other major git hosta are equally tainted- so id suggest using that as the signature list or whatever drama you will do next.

**Comment by lachlanhunt** [comment]
If the Lerna community does decide to adopt a new licence in the future, I would like to encourage you to pick a licence that has been written and reviewed by lawyers.

Regardless of political opinions, ideologies or agendas, copyright licences written by non-lawyers who are unlikely to fully understand the legal ramifications, or otherwise, of the clauses they include are likely not going to be accepted by the wider community of developers; nor the companies they work for that often have internal policies regarding acceptable open source licences for software they can use.

The company I work for, and I can only assume many other companies, are taking a very close look at the licence Jamie wrote to understand the impact upon our own codebases. Lerna just happens to be the most high profile case of it being added, and so far the only case I'm aware of where it got reverted. It also got added to his other repositories and NPM packages, which remain a serious concern.

I would also strongly advise against attempting to use a copyright licence for political protests against contemporary political issues. Software copyright licences are long-lived documents that can outlive any government-of-the-day that happens to be disrupting or upsetting certain groups of people. It's also likely to be an extremely ineffective, if not completely counter-productive, form of protest that can have potentially far reaching consequences beyond the intended goal, including setting a terrible precedent for the industry.

Today, we get a list of companies rejected for working with a government agency carrying out extreme right wing policies. Tomorrow, we get another one from another group of people rejecting companies that support left wing policies. Where does it end?

I urge those in the community advocating for either reinstating the licence or creating a new one in a similar vein to just pause and seriously think about what you're trying to achieve, what you can realistically achieve, and the costs of the way in which you attempt to achieve those goals.

**Comment by lhorie** [comment]
> It's specifically the kind of discussion I feel is needed, with people having to think about these questions discuss and disagree.

@hannahhoward just because there is dissent is not a good enough reason for there to be a discussion. RFCs are not meant to be vehicles for discussion for the sake of discussion. What matters is the arguments for or against a change, its pros and cons and determining actionable steps. A secondary point to consider is what has already happened. Dissenters will dissent and for many topics, there will never be consensus. Discussing endlessly in those cases is not only not fruitful, but distracting as well.

As I understand, the arguments for a modified license are:

- support for the ideology of protesting against ICE, specifically
- desire for a voice in the decision process
- a legitimate desire to use a non-open-source license as a weapon to "punish the bad guys" (a view that seemed to be somewhat trollish in nature, from my observations, but I'm putting it here for completeness)

The arguments for keeping MIT:

- a custom license is restrictive not only to a blacklist, but also any company with a whitelist OSS license policy (i.e. any company of non-trivial size)
- it potentially sets a dangerous precedent that goes against the ideologies of OSS and FSF
- it isn't an appropriate - let alone effective - vehicle for political protest, given that actually disrupting ICE operations could at worst even delay reuniting children with their parents due to potential disruptions in logistics systems

Things I believe should not be part of the discussions re: license:

- whether maintainers with conflict of interest should recuse themselves from discussion - the two most prolific maintainers (other than Jamie) do not have conflict of interests and they agreed on the current decision. Whether maintainers decisions were influenced by one person or another should be of no consequence if they made a decision on their own free will.
- whether maintainers with conflict of interests should have recused themselves prior to the decision - an RFC isn't the place to debate the past, but to determined actionable items (if any) for the future
- whether more voices would change the decision - unless there are different arguments other than the ones I listed above, more commentary at this point is just going to be rehashes of the same ones that were taken into account the first time around.

**Comment by krainboltgreene** [comment]
Only you have suggested that these discussions are "for the sake of discussion".

**Comment by krainboltgreene** [comment]
It's also really weird to see so many people say:

  - It's harmful to [foss|companies|people]
  - I't not [going to work|effective|doing anything]

**Comment by hallister** [comment]
I'll go ahead and suggest this is discussion for the sake of discussion as well. We certainly aren't talking about anything even remotely relevant to Lerna.

**Comment by krainboltgreene** [comment]
Also, @lhorie @hallister is there anything else you'd like to decide we shouldn't discuss? I might as well ask now, since you two are already providing answers.

**Comment by hallister** [comment]
I never said it shouldn't or couldn't be discussed, I said it was discussion for the sake of discussion. Relax buddy, and read what people write before you respond.

**Comment by Sieabah** [comment]
It's still funny the discussion is primarily "If you don't revert it you are complacent with separating families!". Which is a stupid untrue argument.

@krainboltgreene You're providing equal values of _absolutely nothing_ to this discussion.

**Comment by lhorie** [comment]
@krainboltgreene I laid out the factual arguments both pro and against the question of whether Lerna should be MIT or not. I also clearly prefaced my personal opinions as being my opinions. If you're only interested in making ad-hominen arguments, you're not adding to the discussion.

To be clear, my intention was not to dictate, but to spell out reasons why some forms of comments are not fruitful in the context of what an RFC is typically designed for. If it was not clear, I invite you to make new arguments, if you have them, as long as they are on topic.

To that end, I'm going to repeat what I believe is the most important argument to this discussion: if you agree with the tenet that a non-MIT license will have any impact whatsoever on ICE, it follows that doing so may have a _negative_ impact due to its indiscriminate and non-surgical nature of the impact. In other words, disrupting ICE operations can delay families from being reunited.

I don't believe even the most hardcore proponent of the non-MIT license can say in good conscience and good faith that unintended collateral damage is not a real possibility, and that reason alone should be enough for everyone to support the current decision.

**Comment by krainboltgreene** [comment]
You laid out opinions. You can't just declare a comment a "factual argument" and have it be so.

>  just because there is dissent is not enough reason for there to be a discussion

Opinion.

> RFCs are not meant to be vehicles for discussion for the sake of discussion. 

Opinion.

> a legitimate desire to use a non-open-source license as a weapon to "punish the bad guys" (a view that seemed to be somewhat trollish in nature, from my observations, but I'm putting it here for completeness)

Opinion

> a custom license is restrictive not only to a blacklist, but also any company with a whitelist OSS license policy (i.e. any company of non-trivial size)

Opinion.

> whether maintainers with conflict of interest should recuse themselves from discussion

Opinion.

> whether maintainers with conflict of interests should have recused themselves prior to the decision - an RFC isn't the place to debate the past, but to determined actionable items (if any) for the future

Opinion.

> whether more voices would change the decision

Opinion.

**Comment by lhorie** [comment]
> You laid out opinions. You can't just declare a comment a "factual argument" and have it be so.

You laid out opinions. You can't just declare a comment to be a "opinion" and have it be so. Come on, now.

> just because there is dissent is not enough reason for there to be a discussion
> RFCs are not meant to be vehicles for discussion for the sake of discussion.

"An RFC is authored by engineers and computer scientists in the form of a memorandum describing methods, behaviors, research, or innovations applicable to the working of the Internet and Internet-connected systems. It is submitted either for peer review or to convey new concepts, information" [*](https://en.wikipedia.org/wiki/Request_for_Comments)

"Peer review methods are employed to maintain standards of quality, improve performance, and provide credibility"[**](https://en.wikipedia.org/wiki/Peer_review)

> a legitimate desire to use a non-open-source license as a weapon to "punish the bad guys"

I'm more than happy to remove this from the argument list if you think it's not one, though I understand it to be the primary motivation behind the original change by Jamie.

> a custom license is restrictive not only to a blacklist, but also any company with a whitelist OSS license policy (i.e. any company of non-trivial size)

It's a fact that engineers in my company would not be able to use software with such a license, for example. There are precedents documented elsewhere for React's old license as well.

> the rest

Yes, as I said, I prefaced those with "Things _I believe_ should not be part of the discussion". Those are my opinions.

If I may be blunt, it feels like you're acting in bad faith, and I've no interest in drama. @krainboltgreene

**Comment by benwiley4000** [comment]
@SRGOM:

> Can you social justice warriors stop using (and hence promoting) this platform owned by a major ICE collaborator?

![](https://thenib.imgix.net/usq/8688038d-f99b-4224-872b-b8dd626f868c/mister-gotcha-4-9faefa.png?auto=compress,format&cs=srgb&_=9faefab75c06b49cfcf18e1394c50376)

**Comment by robbyoconnor** [comment]
@SRGOM:
> Can you social justice warriors stop using (and hence promoting) this platform owned by a major ICE collaborator?

You really wanna stick it to Microsoft? Delete your GitHub account. Better yet -- move the repository off of GitHub! They're soon going to be its owner.

**Comment by Sieabah** [comment]
I'm confident benwiley4000 and krainboltgreene are just trolling at this point. Don't give them the effort of a response.

**Comment by robbyoconnor** [comment]
Edit: deleted comment on the AWS usage due to not being sure.

**Comment by robbyoconnor** [comment]
The "you're either with us entirely or you're against us" is disgusting. There's also [this](https://twitter.com/webmink/status/1035558595704832000?s=09). I got a similar message from the same guy. Stop this behavior. Grow up. Wanting a better solution doesn't mean we're in favor of what ICE is doing. This change was slacktivism at its finest. Wanna really stick it to MS, get off of GitHub. It was symbolic at best, and even that was kind of laughable.

**Comment by robbyoconnor** [comment]
The "you're either with us entirely or you're against us" is disgusting. There's also [this](https://twitter.com/webmink/status/1035558595704832000?s=09). I got a similar message from the same Ryan White guy. Stop this behavior. Grow up. Wanting a better solution doesn't mean we're in favor of what ICE is doing. This change was slacktivism at its finest. Wanna really stick it to MS, get off of GitHub. Delete your accounts now, Microsoft is going to be the owner of GitHub. It was symbolic at best, and even that was kind of laughable. If you don't delete your GitHub and are still trying to argue that this change being reverted is somehow endorsing what ICE is doing, guess what -- so does having a GitHub account if you're trying to use that argument, your credibility was just shattered.

**Comment by robbyoconnor** [comment]
Also -- if you use Windows -- might I suggest not using it anymore?

**Comment by Sieabah** [comment]
Oh boy...

@termhn The mentions of reverting to the original MIT license has been met with this statement enough times in enough issues, twitter, and reddit. It's not word for word what is said but it is heavily implied by "you support what ICE does".

> I believe this is exactly why we want an RFC -- to discuss whether this change (or one like it) should be implemented or not.

No, it's a terrible idea. Any restriction on the license means it's no longer open source. If that's the goal then I think a lot of people will just drop Lerna because of licensing risk. If that's what you want just to make *a statement* then that's a very poor software decision.

> I don't believe that this intention is at all trollish, though the way you present it makes it sound slightly so.

[It](https://github.com/lerna/lerna/issues/1628#issuecomment-417052979) [was](https://github.com/lerna/lerna/issues/1628#issuecomment-417059200) [entirely](https://github.com/palantir/blueprint/issues/2877#issuecomment-416713800) [a](https://github.com/palantir/tslint/issues/4132#issuecomment-416702198) [trollish](https://github.com/palantir/tslint/issues/4141#issuecomment-416773459) [act](https://github.com/palantir/blueprint/issues/2870#issuecomment-416057140), [it's](https://github.com/palantir/blueprint/issues/2876#issuecomment-416713254) [undebatable](https://github.com/palantir/blueprint/issues/2877#issuecomment-416773124).

> one of which is the license of who gets to use it, in order to affect change in the world. You may say that this is political, but I would argue that by taking no action, you are in fact taking an action and affecting change. We must battle with which one is more positive or has a better ethical standing.

The ethical standing is to honor an open license. If lerna doesn't want to be an open source project anymore that's a different story. You cannot restrict *anyone*. The discussion that everyone seems to be missing is that this change converts Lerna to a proprietary piece of software. Also you're nearing the ethical dilemma of the trolley problem, not exactly, but by saying "inaction is an action in itself" is a  poor argument.

> This assumes that any movement against ICE is inherently bad because it assumes that the short term negative impact would out weigh any long term benefit of having ICE gone. It also ignores the potential ways in which there would be benefits throughout the entire software development and sharing community, as I mentioned near the top of my post.

ICE isn't going anywhere and it probably won't in the next 10 years. Whether that is a positive or negative to whoever reads this is an unfortunate thing we have to deal with. I'm 99.9% sure lerna is nothing special at a government scale, and if ICE (or company in a binding contract with ICE) needed something like lerna they'd build it themselves or simply fork the old version. This license change effectively does nothing to the target. So the discussion is resulting in debating who can virtue signal the strongest.

> I believe that this argument is somewhat of a non-starter, as it asserts that "open source" is fundamentally better than "not open source."

This is a fundamental misunderstanding of what "open" means if one cannot see the issue with a non-open license. You will have to fight the literally brick wall of Stallman to change what "open" source means if you want to challenge this. Non open licenses greatly increase risk of using the software (depending on the license, the bastardization of the MIT-ICE is a huge risk). If you want to build a company you simply don't pick licenses which can destroy your business in a day.

> I think this is somewhat silly however, as what is and is not open source has been malleable and changed before, and I don't think it is out of the question for it to change again.

In what realm of reality does a **strictly** always open door, close? Open is open, closed is closed. Open source is non discriminatory, if you want to go down the path of "restrictive licensing is open source" then I don't see the issue with Facebook being able to restrict the people using React under their Patent license. Absolutely zero issues with a single entity controlling who uses the software while claiming to be open source.

**Comment by benwiley4000** [comment]
Folks who use terms like "virtue signaling" might not realize that is incredibly politically charged.

Also it's simply not true that restrictive clauses have no precedent in open source work. Everyone screaming that "this is no longer open source" needs to realize that A) we are not beholden to the OSI's definition of open source, and B) there is plenty of precedent with Commons Clause, Fair Source, License Zero, and even the original license for JSON (thanks to Mikeal Rogers [formerly of the Node.js Foundation] for pointing that out on Twitter). ([source](https://twitter.com/mikeal/status/1034924182126047232?s=19)) ([source](https://twitter.com/mikeal/status/1035173364355162112?s=19))

**Comment by krainboltgreene** [comment]
So we neither have a compelling reason for Jamie K. to have been kicked nor a compelling reason why the license was reverted. I am *shocked*.

So to recap:

  - All maintainers accepted the original pull request.
  - There's absolutely prior art on doing this sort of license change.
  - Jamie K.'s behavior was apparently considered bad, but not until big companies were involved.
  - No one in charge of this has listed any behavior that violated in the current implemented code of conduct.
  - We're still waiting on an official response to this issue, *which was created because of (what is now) a clear attempt at silencing*.
  - Multiple employees from a company on the list swooped down to convince a maintainer to revert the change (but not before specifically moving to revert their company's inclusion in the list, *awkward*).

And the result?

  - An immense amount of character assassination
  - The gamergate, anti-coc, and hard right crowd have been riled up
  - A fork that will surely not see life longer than a week but will have eaten plenty of valuable engineering time
  - A new surge of engineers who think ethics aren't their problem

**Comment by Sieabah** [comment]
The comments he made on other projects is a violation of the CoC and that's why he was removed. I think that may have been the nth time someone has mentioned that to you. 

So either you're illiterate or have an agenda.

**Comment by hedgepigdaniel** [comment]
I'll just point out a particular sentence from the original licence change: https://github.com/lerna/lerna/pull/1616

> Recently it has come to my attention that many of these companies which are being paid millions of dollars by ICE are also using some of the open source software that I helped build.

Emphasis on "helped"

I think there's quite a bit of needless character assassination going on. Perhaps Jamie has made some unfortunate statements at different times, but in the context of this issue, it hardly looks like he has claimed personal ownership of Lerna or any other open source project.

**Comment by Sieabah** [comment]
@hedgepigdaniel He did.

Here is one where he directly states Lerna is his project. https://github.com/palantir/blueprint/issues/2877
Here is one where he claims babel is his project https://github.com/palantir/tslint/issues/4132
Another one https://github.com/palantir/tslint/issues/4141
Why does it keep happening D:? https://github.com/palantir/blueprint/issues/2870

There is no character assassination going on, he acted shitty and is being called out on it.

**Comment by robbyoconnor** [comment]
@SRGOM -- no I said for people who are against the change being reverted to look at the fact that they are using MS products -- which makes them hypocrites. If you want to boycott Microsoft, maybe don't use their products -- or even better get off of GitHub.

**Comment by robbyoconnor** [comment]
@SRGOM -- So for all this grandstanding you're doing -- why not delete your GitHub account...seriously -- MS will own GitHub soon. Also -- change your OS from Windows -- if you use that...

**Comment by lhorie** [comment]
> I believe this is exactly why we want an RFC -- to discuss whether this change (or one like it) should be implemented or not

I agree that this is the intent, but in practice, that's not the impression I got from reading this thread so far. The majority of comments are either trolling or doing passive-aggressive witch hunts (and sadly I'm seeing this behaviour from both sides of the discussion, mind you).

I'm not sure if there needs to be more clarity in terms of an RFC is typically supposed to look like (given that it is, admittedly, a new process in this specific org), which is why I offered my opinions on what is appropriate. If people disagree, they could potentially look at other orgs' RFC processes to get an idea of how one is structured (React and Ember both have RFC processes that could serve as reference, for example). In any case, I think we would all agree that all commenters should be following the basic rule of thumb of "if a comment would be inappropriate in a technical discussion, it has no place in an RFC discussion either".

> The idea is that we can use software and the control over we have over the software we write, one of which is the license of who gets to use it

I understand this intent, but I have not seen anyone address the concerns about what happens with  non-blacklisted companies being unable to use such licensed software was also dismissed as if it were a problem that did not exist, rather than a significant one (especially for something that is so geared towards big company workflows such as Lerna). For example, @evocateur (or whoever is the maintainer at any time)'s employer could disallow him from using a hypothetically non-OSI-compliant Lerna at work. If that was the only reason he was working on Lerna, there's a good chance the project would lose its steward and effectively die. Likewise, I might contribute a PR to a MIT-licensed Lerna if I need the fix for work, but I will certainly not do so if I'm not allowed to use it due to licensing issues in the first place.

> This assumes that any movement against ICE is inherently bad because it assumes that the short term negative impact would out weigh any long term benefit of having ICE gone

No, the point is that we _can't possibly estimate the impact_, both positive and negative. We cannot make assumptions that the positives will outweigh the negatives or vice versa. The argument for a non-OSS license boils down to "something must be done, this is something, therefore it must be done". No, it must not. Whatever was the latest heated Javascript-related topic of the week on Hacker News is statistically unlikely to be even remotely close to being an appropriate response to an issue that likely no one here has any real life understanding of, other than what they read on some news site while sipping their coffees.

It's especially disingenuous to say this must be done, if the person saying so did not even bother to try to research what other things could have been done instead, or what is being done, even. To illustrate the level of myopia here, there was a PR on Jamie's repo to add Uber to the blacklist, despite the fact that Uber is _donating rides and meals to families separated by ICE_[*](https://thepointsguy.com/news/uber-donating-rides-separated-families/).

> it asserts that "open source" is fundamentally better than "not open source."

As I mentioned in an earlier comment, there's already a well established way of running non-OSS software: pay for it. This very discussion is open because the maintainers feel that inclusion of all parties is of utmost importance to the success of this project. They could just tell everyone to go away, or even pull a uws[**](https://www.reddit.com/r/node/comments/91kgte/uws_has_been_deprecated/) and _they'd be entirely within their rights to do so_. Using the privilege of having this forum to argue that the maintainers should adopt a restrictive license is biting the hand that feeds. I think the non-starter here is to suggest that maintainers ought to maintain a project that they themselves may not be able to use under some circumstances.

**Comment by benwiley4000** [comment]
@robbyoconnor you and @SRGOM agree, they were being sarcastic......

**Comment by hedgepigdaniel** [comment]
@Sieabah as I said perhaps he made some unfortunate statements - what I'm trying to say is that if you look at the license change PR where he elaborated, it does not look like he thinks he personally owns the project at the exclusion of anybody else. He did not push the change without review. When people say "my" thing, it can mean a lot of different things - its usually not helpful to draw strong conclusions from the use of that word.

Looking at the commit history, it does look like he was heavily involved in the early days of Lerna. I assume he means "my" in a similar sense - "Lerna is a project which I was heavily involved in and I share significant responsibility for creating".

**Comment by robbyoconnor** [comment]
> @robbyoconnor you and @SRGOM agree, they were being sarcastic......

Oh Jesus nevermind -- I'm an idiot I guess :joy: -- apologies @SRGOM :)

**Comment by termhn** [comment]
@lhorie 

> I'm not sure if there needs to be more clarity in terms of an RFC is typically supposed to look like (given that it is, admittedly, a new process in this specific org), which is why I offered my opinions on what is appropriate. 

I would agree if this were the actual RFC thread, but this is actually just a thread requesting an RFC should exist in the first place. Therefore I don't think it's valid to judge peoples' comments here as what they would have said in an actual RFC.

>Whatever was the latest heated Javascript-related topic of the week on Hacker News is statistically unlikely to be even remotely close to being an appropriate response to an issue that likely no one here has any real life understanding of, other than what they read on some news site while sipping their coffees.

And yet the reason we have OSS licenses as a staple of modern software development today was the continued preaching of outcast mailing list users in the last few decades, and you could extend this to basically any successful political protest as well.

>As I mentioned in an earlier comment, there's already a well established way of running non-OSS software: pay for it.

There's also lots of not OSS software that is free as in beer.

> This very discussion is open because the maintainers feel that inclusion of all parties is of utmost importance to the success of this project.

And I appreciate that this is true, but being "non-OSS" would not preclude this from happening by default anyway.

@Sieabah 
> This is a fundamental misunderstanding of what "open" means if one cannot see the issue with a non-open license. You will have to fight the literally brick wall of Stallman to change what "open" source means if you want to challenge this.

And yet Stallman's own license is one of the most restrictive and closed-door OSS licenses there is. He just closes the door on a different group of people: the group of people that want to use or modify his code without sharing back their changes. There's not a fundamental difference between that and precluding groups for another reason. This is also why I think the OSI page on "evil", originated from the JSON license, is fairly crap. I think the "do no evil line" is also terrible because it's even less enforceable than what the MIT-ICE license was, but the discussion in OSS around being able to preclude wrongdoers from using code has been very backwards IMO.

> ICE isn't going anywhere and it probably won't in the next 10 years. Whether that is a positive or negative to whoever reads this is an unfortunate thing we have to deal with. I'm 99.9% sure lerna is nothing special at a government scale, and if ICE (or company in a binding contract with ICE) needed something like lerna they'd build it themselves or simply fork the old version. This license change effectively does nothing to the target. 

As I said, Lerna's own license change is not meant to harm ICE by itself in any significant way, it's meant to

1. Deal with the ethical concerns of allowing a harmful entity to use your code
2. Show other OSS projects that this is a real dilemma and hopefully spark them to discuss and maybe take the same action

If many OSS projects do the same thing, then there could be real, large-scale change over time.

> It was entirely a trollish act, it's undebatable. 👈 Multiple links

None of those show a trollish intent behind the action at all, and in the actual PR that was made, the changes were discussed in a very professional manner, all (at the time available) protocols were followed, and a calm and deliberate reasoning was given. Being terse and (arguably) rude to companies over their using tools he contributed to does not constitute the action being trollish.

**Comment by Sieabah** [comment]
@hedgepigdaniel Sure, I'm sure you could argue hitler was a pretty cool guy if you ignore all the other stuff he did and focuses on his gun control policy. 🔥

If you just excuse it as "unfortunate" then you're just trying to justify your own opinions.

@termhn The difference between having a strict license vs claiming it's open. As I said numerous times, if the goal is to have lerna be proprietary that's a different discussion. At the current time, and as far as I know, lerna plans to stay as an *open* source project.

> If many OSS projects do the same thing, then there could be real, large-scale change over time.

Then none of them are open source anymore. You have collectively made a proprietary license. We're not arguing that so it's a moot point to even mention.

> None of those show a trollish intent behind the action at all, and in the actual PR that was made, the changes were discussed in a very professional manner, all (at the time available) protocols were followed, and a calm and deliberate reasoning was given.

He made multiple issues in the same project that all state the same thing, linking to the same resources which are there to incite hatred from a group of people towards a company. See this definition:

> trolling: make a deliberately offensive or **provocative** online post with the aim of upsetting someone or eliciting an angry response from them.

My god, it's exactly what he did.

**Comment by evocateur** [comment]
And we're done. I'm going to spend the rest of the long weekend restoring my mental health. I'll resume my pariah duties on Tuesday.
